controllers/user.js

```diff
@@ -70,20 +141,49 @@ router.put( '/:id/lexemes', (req, res, next) => {
       // persist in DB, deliver response
       req.user.update( { lexemes: [...lexemesFiltered, ...lexemesToAdd] }, (error, result) => {
         if ( error ) return next(error);
-        res.json(result); // just "res.status(200).send()" ?
+        res.json(result); // unpopulated! just "res.status(200).send()" ?
       });
 
     })
     .catch( error => next(error) );
 });
 
 
+
+
 router.get( '/:id/lexemes/:quantity', (req, res, next) => {
   User.findById( req.user._id )
     .populate('lexemes.lexeme')
     .then( ({ lexemes }) => {
       // select questions, later based on (1) level of control, (2) last learned
-      return res.json(lexemes);
+
+      const filteredLexemes = lexemes
+        .filter( userLexeme => { // check if there is at least 1 task for requested level
+          return userLexeme.lexeme.tasks.find( task => task.level === req.user.level ) !== undefined;
```

Comment:
No need to check if is not `undefined` instead:
`!!userLexeme.lexeme.tasks.find( task => task.level === req.user.level );` so you do a cast to `boolean`